test/integration/overload_integration_test.cc

```diff
@@ -1,58 +1,126 @@
+#include <unordered_map>
+
 #include "envoy/config/bootstrap/v3/bootstrap.pb.h"
 #include "envoy/config/overload/v3/overload.pb.h"
+#include "envoy/server/resource_monitor.h"
+#include "envoy/server/resource_monitor_config.h"
 
+#include "test/common/config/dummy_config.pb.h"
 #include "test/integration/http_protocol_integration.h"
+#include "test/test_common/registry.h"
 
 #include "absl/strings/str_cat.h"
 
 namespace Envoy {
 
+class FakeResourceMonitorFactory;
+
+class FakeResourceMonitor : public Server::ResourceMonitor {
+public:
+  static constexpr absl::string_view kName =
+      "envoy.resource_monitors.testonly.fake_resource_monitor";
+
+  FakeResourceMonitor(Event::Dispatcher& dispatcher, FakeResourceMonitorFactory& factory)
+      : dispatcher_(dispatcher), factory_(factory), pressure_(0.0) {}
+  ~FakeResourceMonitor() override;
+  void updateResourceUsage(Callbacks& callbacks) override;
+
+  void setResourcePressure(double pressure) {
+    dispatcher_.post([this, pressure] { pressure_ = pressure; });
+  }
+
+private:
+  Event::Dispatcher& dispatcher_;
+  FakeResourceMonitorFactory& factory_;
+  double pressure_;
+};
+
+class FakeResourceMonitorFactory : public Server::Configuration::ResourceMonitorFactory {
+public:
+  FakeResourceMonitor* monitor() const { return monitor_; }
+  Server::ResourceMonitorPtr
+  createResourceMonitor(const Protobuf::Message& config,
+                        Server::Configuration::ResourceMonitorFactoryContext& context) override;
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return std::make_unique<test::common::config::DummyConfig>();
+  }
+
+  std::string name() const override { return std::string(FakeResourceMonitor::kName); }
```

Comment:
nit: Consider changing to:
std::string name() const override { return "envoy.resource_monitors.testonly.fake_resource_monitor"; }

There are no other uses of kName